src/lib/utils.js

```diff
@@ -123,18 +123,27 @@ function prepareTemplates(defaultTemplates, templates) {
   }, {templates: {}, useCustomCompileOptions: {}});
 }
 
-function getRefinement(attributeName, name, resultsFacets) {
-  let res = {attributeName, name};
-  let facet = find(resultsFacets, (_facet) => { return _facet.name === attributeName; });
-  if (facet !== undefined) {
-    const count = get(facet, 'data.' + name);
-    const exhaustive = get(facet, 'exhaustive');
-    if (count !== undefined) {
-      res.count = count;
-    }
-    if (exhaustive !== undefined) {
-      res.exhaustive = exhaustive;
+function getRefinement(state, type, attributeName, name, resultsFacets) {
+  let res = {type, attributeName, name};
+  let facet = find(resultsFacets, {name: attributeName});
+  let count;
+  if (type === 'hierarchical') {
+    let facetDeclaration = state.getHierarchicalFacetByName(attributeName);
+    let splitted = name.split(facetDeclaration.separator);
+    res.name = splitted[splitted.length - 1];
```

Comment:
I wonder if you could reach this more easily: https://github.com/algolia/algoliasearch-helper-js/blob/master/README.md#asking-for-the-current-breadcrumb